superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -321,12 +354,7 @@ export default class ResultSet extends React.PureComponent<
   getUserDatasets = async (searchText = '') => {
     // Making sure that autocomplete input has a value before rendering the dropdown
     // Transforming the userDatasetsOwned data for SaveModalComponent)
-    const appContainer = document.getElementById('app');
-    const bootstrapData = JSON.parse(
-      appContainer?.getAttribute('data-bootstrap') || '{}',
-    );
-
-    if (bootstrapData.user && bootstrapData.user.userId) {
+    if (this.props.user?.userId) {
```

Comment:
since you use this value twice here, do you want to destructure just the userId out?